### PR TITLE
setup the npm registry url with setup-node config

### PR DESCRIPTION
**Pull request by threepointone**
This should remove the `WARN  Issue while reading "/Users/***/dev/cloudflare/workers-sdk/.npmrc". Failed to replace env in config: ${NPM_PUBLISH_TOKEN} ` warning we always

### CODE REVIEW

File: .github/workflows/c3-e2e.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
       - name: E2E Tests
         uses: ./.github/actions/run-c3-e2e
         env:
-          NPM_PUBLISH_TOKEN: NOT_USED
+          NODE_AUTH_TOKEN: NOT_USED
```
Comment (petebacondarwin) [inline .github/workflows/c3-e2e.yml:57]: Is this needed now?

Comment (petebacondarwin) [reply .github/workflows/c3-e2e.yml:57]: see https://github.com/actions/setup-node/blob/eff380dfbcf941bf8832e4acb788cebe13dfd758/src/authutil.ts#L54

Comment (threepointone) [reply .github/workflows/c3-e2e.yml:57]: probably not, lemme remove it

**Review by petebacondarwin — APPROVED** [review]
```
       .....           .....
   ,ad8PPPP88b,     ,d88PPPP8ba,
  d8P"      "Y8b, ,d8P"      "Y8b
 dP'           "8a8"           `Yd
 8(              "              )8
 I8                             8I
  Yb,                         ,dP
   "8a,                     ,a8"
     "8a,                 ,a8"
       "Yba             adP"
         `Y8a         a8P'
           `88,     ,88'
             "8b   d8"
              "8b d8"
               `888'
                 "
```

**Comment by threepointone** [comment]
NO U

**Pull request merged**